### PR TITLE
chore: Integrate authentication library

### LINKED ISSUE #2: Integrate authentication library
Using https://github.com/nearform/fastify-auth0-verify. See: https://github.com/nearform/fastify-auth0-verify/issues/224

Integrate in every endpoint that was implemented before this work.

### PR DESCRIPTION
**Pull request by gnarea**
Fixes #2.

High-level changes:

- Integrate [`mock-oauth2-server`](https://github.com/navikt/mock-oauth2-server) in development.
- Integrate fork of [`fastify-auth0-verify`](https://github.com/nearform/fastify-auth0-verify). See #82.

#84 will actually use `request.user` to enforce authorisation, so no endpoint is being updated to enforce authorisation here.

# Testing

This PR can be verified by creating the dummy route `GET /authn` as follows:

```diff
diff --git a/src/services/routes/healthcheck.routes.ts b/src/services/routes/healthcheck.routes.ts
index 7b7abe7..6cf48d3 100644
--- a/src/services/routes/healthcheck.routes.ts
+++ b/src/services/routes/healthcheck.routes.ts
@@ -20,5 +20,15 @@ export default function registerRoutes(
     },
   });
 
+  fastify.route({
+    method: ['GET'],
+    url: '/authn',
+    preValidation: fastify.authenticate,
+
+    async handler(request, reply): Promise<void> {
+      await reply.code(HTTP_STATUS_CODES.OK).send(request.user);
+    },
```

Then making requests to it as follows:

```http
### Make anonymous request
GET http://veraid-authority.default.10.103.177.106.sslip.io/authn

### Make authenticated request with invalid credentials
GET http://veraid-authority.default.10.103.177.106.sslip.io/authn
Authorization: Bearer INVALID_TOKEN

### Make authenticated request with valid credentials
GET http://veraid-authority.default.10.103.177.106.sslip.io/authn
Authorization: Bearer <VALID-TOKEN-HERE>
```

`<VALID-TOKEN-HERE>` can be obtained with:

```http
### Authenticate with authorisation server (client credentials)
POST http://mock-authz-server.default.10.103.177.106.sslip.io/default/token
Content-Type: application/x-www-form-urlencoded

grant_type=client_credentials&client_id=admin@example.com&client_secret=s3cr3t
```

### CODE REVIEW

File: src/services/fastify.ts [hunk 1]
```diff
@@ -20,6 +21,26 @@ const DEFAULT_REQUEST_ID_HEADER = 'X-Request-Id';
...
+    (issuer === undefined && issuerRegex === undefined) ||
+    (issuer !== undefined && issuerRegex !== undefined)
+  ) {
+    throw new Error('Either OAUTH2_TOKEN_ISSUER or OAUTH2_TOKEN_ISSUER_REGEX should be set');
+  }
+
+  return {
+    audience,
+    domain,
+    issuer: issuer ?? (issuerRegex as any),
```
Comment (gnarea) [inline src/services/fastify.ts:40]: The `as any` is a workaround for https://github.com/nearform/fastify-auth0-verify/pull/279

File: package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
     "@typegoose/typegoose": "^10.3.3",
     "env-var": "^7.3.0",
     "fastify": "^4.14.0",
+    "fastify-auth0-verify": "github:relaycorp/fastify-auth0-verify#auth0-agnostic-generated",
```
Comment (levansuper) [inline package.json:28]: Why are we forking this?

Comment (levansuper) [reply package.json:28]: I guess we will switch back to the original when the issue is merged?

Comment (gnarea) [reply package.json:28]: Both questions are answered in the PR description:

> Integrate fork of [fastify-auth0-verify](https://github.com/nearform/fastify-auth0-verify). See https://github.com/relaycorp/veraid-authority/issues/82.

Comment (levansuper) [reply package.json:28]: Yeah I saw that. checked the code first

**Review by levansuper — APPROVED** [review]

**Pull request merged**